prow/plugins/lgtm/lgtm.go

```diff
@@ -300,12 +333,49 @@ func handlePullRequest(gc ghLabelClient, pe github.PullRequestEvent, log *logrus
 		return nil
 	}
 
-	// Don't bother checking if it has the label...it's a race, and we'll have
-	// to handle failure due to not being labeled anyway.
 	org := pe.PullRequest.Base.Repo.Owner.Login
 	repo := pe.PullRequest.Base.Repo.Name
 	number := pe.PullRequest.Number
 
+	// Check if we have LGTM label
+	hasLGTM := false
+	labels, err := gc.GetIssueLabels(org, repo, number)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get labels.")
+	}
+	hasLGTM = github.HasLabel(LgtmLabel, labels)
+
+	if hasLGTM == true {
+		// Check if we have a tree-hash comment
+		var lastLgtmTreeHash string
+		botname, err := gc.BotName()
+		if err != nil {
+			return err
+		}
+		comments, err := gc.ListIssueComments(org, repo, number)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to get issue comments.")
+		}
+		for _, comment := range comments {
+			m := addLGTMLabelNotificationRe.FindStringSubmatch(comment.Body)
+			if comment.User.Login == botname && m != nil {
+				lastLgtmTreeHash = m[1]
+				break
+			}
+		}
+		// Get the current tree-hash
+		commit, err := gc.GetSingleCommit(org, repo, *pe.PullRequest.MergeSHA)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to get commit.")
+		}
+		treeHash := commit.Commit.Tree.SHA
+		if treeHash == lastLgtmTreeHash {
+			// Don't remove the label, PR code hasn't changed
+			log.Infof("Keeping LGTM label as the tree-hash remained the same: %s", treeHash)
+			return nil
+		}
+	}
+
 	var labelNotFound bool
 	if err := gc.RemoveLabel(org, repo, number, LgtmLabel); err != nil {
```

Comment:
We should not try to remove the label if `!hasLGTM`. You can move the contents of the `if hasLGTM == true {` block outside the conditional and instead first check if `!hasLGTM` and return immediately if so.